### PR TITLE
(PCP-57) Determine default path for modules dir

### PR DESCRIPTION
**Pull request by parisiale**
Here if set the default path for the modules dir: on *nix we use the
location on the /opt/.. tree specified in Puppet Specs; for Windows we
use the relative path of the pxp-agent.exe executable.

### CODE REVIEW

**Comment by parisiale** [comment]
@MikaelSmith @Iristyle @kylog  github automatically closed the PR #123, when I force pushed the new branch.

**Comment by parisiale** [comment]
Tested on Windows 2012r2. The help text shows the following:

'''
PS C:\pxp-agent\release> pwd

Path
----
C:\pxp-agent\release


PS C:\pxp-agent\release> .\bin\pxp-agent.exe --help

Usage: pxp-agent [options]

Options:
   -h
  --help                      Show this message
  --verbose                   Set verbose output
  --version                   Display version information
  --ca <str>                  CA certificate
  --cert <str>                pxp-agent certificate
  --config-file <str>         Config file, default:
                              C:\ProgramData\PuppetLabs\pxp-agent\etc\pxp-agent.conf
  --console-logger            Show log messages only on console, default: false
                              (log to file)
  --foreground                Don't daemonize, default false
  --key <str>                 pxp-agent private key
  --logdir <str>              Log directory, default:
                              C:\ProgramData\PuppetLabs\pxp-agent\var\log
  --loglevel <str>            Log level; defaults to 'info'
  --modules-config-dir <str>  Module config files directory, default:
                              C:\ProgramData\PuppetLabs\pxp-agent\etc\modules
  --modules-dir <str>         Modules dirctory, default (relative to the
                              pxp-agent.exe path): C:\pxp-agent\release\modules
  --server <str>
   -s <str>                   PCP server URL
  --spool-dir <str>           Spool action results directory, default:
                              C:\ProgramData\PuppetLabs\pxp-agent\var\spool
'''

**Comment by Iristyle** [comment]
:+1:

File: lib/src/configuration.cc [hunk 1]
```diff
@@ -44,19 +44,34 @@ namespace lth_log = leatherman::logging;
     const std::string DEFAULT_SPOOL_DIR { (DATA_DIR / "var" / "spool").string() };
     const std::string PID_DIR { (DATA_DIR / "var" / "run").string() };
     static const std::string DEFAULT_LOG_DIR { (DATA_DIR / "var" / "log").string() };
+
+    static const std::string DEFAULT_MODULES_DIR = []() {
+        wchar_t szPath[MAX_PATH];
+        if (GetModuleFileNameW(NULL, szPath, MAX_PATH) == 0) {
+            throw Configuration::Error { "failed to retrive pxp-agent binary path" };
+        }
+        // Go up twice, assuming the subtree from Puppet Specs:
```
Comment (MikaelSmith) [inline lib/src/configuration.cc]: Should this also be made to work with the normal make install? I guess we don't want to have a generic modules directory, so I don't see that working.

**Pull request merged**